app/controllers/taxonomy_controller.rb

```diff
@@ -33,13 +32,13 @@ def nest_categories_children(categories)
       category_index[category[:id]] = category
     end
     categories.each do |category|
-      if category.parentCategory
-        parent = category_index[category.parentCategory]
+      category[:parentCategory].each do |parent_id|
+        parent = category_index[parent_id]
         parent[:children] ||= []
         parent[:children] << category
       end
     end
-    categories.reject! { |category| category.parentCategory }
+    categories.reject! { |category| category[:parentCategory].any? }
```

Comment:
is it not better like this `categories.select! { |category| category[:parentCategory].blank? }`, but not sure to test.

Reply:
It's the inverse actually (delete the categories that have at least one parent)
and I didn't want to do `!category[:parentCategory].blank?`

Reply:
I'm not sure that `any?` does work the way you would expect, but maybe I'm wrong

Reply:
but let's assume it is for now

Reply:
ok. maybe you just need to add `&` to safe check nil
![image](https://github.com/user-attachments/assets/2a66c840-3bb8-470e-9773-7c0981bb8a5d)

Reply:
Done